src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/CreatePluginHandlerImpl.java

```diff
@@ -0,0 +1,238 @@
+ */
+
+package com.epam.ta.reportportal.core.integration.plugin.impl;
+
+import com.epam.ta.reportportal.commons.validation.BusinessRule;
+import com.epam.ta.reportportal.commons.validation.Suppliers;
+import com.epam.ta.reportportal.core.integration.plugin.CreatePluginHandler;
+import com.epam.ta.reportportal.core.integration.plugin.PluginInfo;
+import com.epam.ta.reportportal.core.integration.plugin.PluginLoader;
+import com.epam.ta.reportportal.core.integration.util.property.IntegrationDetailsProperties;
+import com.epam.ta.reportportal.core.integration.util.property.ReportPortalIntegrationEnum;
+import com.epam.ta.reportportal.core.plugin.PluginBox;
+import com.epam.ta.reportportal.dao.IntegrationTypeRepository;
+import com.epam.ta.reportportal.entity.integration.IntegrationType;
+import com.epam.ta.reportportal.entity.integration.IntegrationTypeDetails;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.epam.ta.reportportal.filesystem.DataStore;
+import com.epam.ta.reportportal.core.integration.plugin.PluginUploadingCache;
+import com.epam.ta.reportportal.ws.model.EntryCreatedRS;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import org.apache.commons.lang3.StringUtils;
+import org.pf4j.PluginWrapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.validation.constraints.NotNull;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Optional;
+
+import static com.epam.ta.reportportal.commons.Predicates.notNull;
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@Service
+public class CreatePluginHandlerImpl implements CreatePluginHandler {
+
+	public static final String PLUGIN_TEMP_DIRECTORY = "/temp/";
+
+	private final String pluginsRootPath;
+
+	private final PluginBox pluginBox;
+
+	private final PluginLoader pluginLoader;
+
+	private final IntegrationTypeRepository integrationTypeRepository;
+
+	private final DataStore dataStore;
+
+	private final PluginUploadingCache pluginUploadingCache;
+
+	@Autowired
+	public CreatePluginHandlerImpl(@Value("${rp.plugins.path}") String pluginsRootPath, PluginBox pluginBox, PluginLoader pluginLoader,
+			IntegrationTypeRepository integrationTypeRepository, DataStore dataStore, PluginUploadingCache pluginUploadingCache) {
+		this.pluginsRootPath = pluginsRootPath;
+		this.pluginBox = pluginBox;
+		this.pluginLoader = pluginLoader;
+		this.integrationTypeRepository = integrationTypeRepository;
+		this.dataStore = dataStore;
+		this.pluginUploadingCache = pluginUploadingCache;
+	}
+
+	@Override
+	public EntryCreatedRS uploadPlugin(MultipartFile pluginFile) {
```

Comment:
probably we need to create more private methods to clean the method up

Reply:
done